code/modules/mob/living/carbon/carbon.dm

```diff
@@ -77,7 +77,7 @@
 		if(!hurt)
 			return
 
-		if(victim.check_block(src, 0, "[name]", LEAP_ATTACK))
+		if(victim.check_block(src, 0, "[name]", LEAP_ATTACK) || . == SUCCESSFUL_BLOCK)
```

Comment:
I feel like we should check that parent call was blocked before running check block, so we don't needlessly call check block if we know it already is 
```suggestion
		if(. == SUCCESSFUL_BLOCK || victim.check_block(src, 0, "[name]", LEAP_ATTACK))
```